runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -287,6 +286,8 @@ public class StreamingDataflowWorker {
   // Possibly overridden by streaming engine config.
   private int maxWorkItemCommitBytes = Integer.MAX_VALUE;
 
+  private DataflowExecutionStateSampler sampler = DataflowExecutionStateSampler.instance();
```

Comment:
can be final if not reassigned.

also is `DataflowExecutionStateSampler` a singleton?

Reply:
3be2aa33f9753b5bad6e50190986ee3a62763967

we're using it as a singleton for the StreamingDataflowWorker but the base class is not necessarily intended to be a singleton